### PR TITLE
Implement vulkan video extensions

### LINKED ISSUE #722: Safe wrapper for the video extension
Hello everyone, as a member of vulkan video working group under Khronos I would like contribute to this project by providing safe wrapper for the video extension. Spec for the encode is still in flux but I volunteer to ensure compatibility in future releases.

For I while Im have ash fork with added extensions:
https://github.com/neurotok/ash.git
And a basic encode example:
https://github.com/neurotok/ash-video.git

But the problem which I cannot overcome is a panic when calling even basic functions from the encode:
https://docs.rs/ash/latest/ash/vk/struct.KhrVideoQueueFn.html#
(using my ash-video example)

I had some success when I wrote this wrapper as an instance extension, but I'm running out of ideas on how the wrapper should be designed as device one. 

@MarijnS95 , @MaikKlein ? Could you please take a look at my ash fork and provided example ?

### PR DESCRIPTION
**Pull request by Cach30verfl0w**
Closes #722

Added support for the video extensions.

### CODE REVIEW

File: ash/src/extensions/khr/video_queue.rs [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+        out: &mut [vk::VideoFormatPropertiesKHR<'_>],
+    ) -> VkResult<()> {
+        let mut count = out.len() as u32;
+        (self.fp.get_physical_device_video_format_properties_khr)(
+            physical_device,
+            video_format_info,
+            &mut count,
+            out.as_mut_ptr(),
+        ).result()
+    }
```
Comment (MarijnS95) [inline ash/src/extensions/khr/video_queue.rs:59]: Missing an `assert` that `count` wasn't updated to value `!= out.len()`.

File: ash/src/extensions/khr/video_queue.rs [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+        allocation_callbacks: Option<&vk::AllocationCallbacks<'_>>,
+    ) {
+        (self.fp.destroy_video_session_khr)(
+            self.handle,
+            video_session,
+            allocation_callbacks.as_raw_ptr(),
+        )
+    }
+
+    // Retrieve the number of elements to pass to [`get_video_session_memory_requirements`][Self::get_video_session_memory_requirements]
```
Comment (MarijnS95) [inline ash/src/extensions/khr/video_queue.rs:94]: Same as above.

File: ash/src/extensions/khr/video_queue.rs [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+        out: &mut [vk::VideoSessionMemoryRequirementsKHR<'_>],
+    ) -> VkResult<()> {
+        let mut count = out.len() as u32;
+        (self.fp.get_video_session_memory_requirements_khr)(
+            self.handle,
+            video_session,
+            &mut count,
+            out.as_mut_ptr(),
+        ).result()
+    }
```
Comment (MarijnS95) [inline ash/src/extensions/khr/video_queue.rs:127]: Same assert is missing here.

File: ash/src/extensions/khr/video_queue.rs [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+    #[inline]
+    pub unsafe fn bind_video_session_memory(
+        &self,
+        video_session: vk::VideoSessionKHR,
+        bind_session_memory_infos: &[vk::BindVideoSessionMemoryInfoKHR<'_>],
+    ) -> VkResult<()> {
+        (self.fp.bind_video_session_memory_khr)(
+            self.handle,
+            video_session,
+            bind_session_memory_infos.len() as _,
```
Comment (MarijnS95) [inline ash/src/extensions/khr/video_queue.rs:139]: Only `khr/acceleration_structures.rs` is erroneously using typeless casts. Please specify `u32` here explicitly.

**Comment by Cach30verfl0w** [comment]
I have new hardware where I cannot use and test the encoding capabilities. Accordingly, I am closing this PR as I unfortunately cannot work on these here.

**Pull request closed without merging**